src/main/php/util/data/Sequence.class.php

```diff
@@ -42,18 +42,21 @@ protected function __construct($elements) {
    *
    * @param  function(): var $operation
    * @return var
+   * @throws util.data.CannotReset
    */
   protected function terminal($operation) {
     static $message= 'Underlying value is streamed and cannot be processed more than once';
```

Comment:
That's right, I'll probably write a bunch of test code and see where that gets me.